### DISCUSSION TITLE
Z Axis Align with UBL?

### DISCUSSION THREAD

**Discussion by names12** [Jyers/Marlin#1199] [Q&A]
Hey All, 

My question: will running G34 work with G29 and if so, do I run the G34 before or after G29? Will G29 override the G34 or vise versa? 

Details: I just recently upgraded my E3V2 with a direct drive and dual lead screws. After my first set of testing, I'm seeing some Z-banding that is a bit worse than it was before the upgrades. I learned that I need to align the z-axis by running G34 in the start code. 
That's fine, but I'm currently running a Jyers UBL with BLTouch 10x10 firmware with G29 L0 (loads the mesh in slot 0) and G29 J2 which probes the corners of the bed all for tilting the mesh. Everything else is great! The UBL is functioning perfectly and prints are looking awesome, except for the z-banding issue. 

Any help is greatly appreciated! Thank you!

**Comment by BlueWings172** [comment]
@tititopher68-dev 

where do you get the sync kit from ? I have dual Z kit (2 motors) from Creality which of course does not include the belt mechanism.

So how to have 2 motors + belt system ? I only find kits that either have belt or motor.

I have some z-banding that I can't solve.

**Comment by BlueWings172** [comment]
Looking at the dual z kits (1 motor) on amazon, they are quite expensive now reaching over $50 (good rated ones) and my kit (2 motor) includes many of the parts in the other kit so I can't justify the cost. I'm thinking to go your route and DIY it.

By helical screws, do you mean the T8 lead screws ? if not, did you buy new longer lead screws because it seems the stock ones are a bit short?
